x-pack/legacy/plugins/canvas/canvas_plugin_src/strings/ui.ts

```diff
@@ -0,0 +1,715 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { CSS, URL, MARKDOWN } from '../../i18n';
+
+export const ArgumentStrings = {
+  FilterGroup: {
+    getDisplayName: () =>
+      i18n.translate('xpack.canvas.uis.arguments.filterGroupTitle', {
+        defaultMessage: 'Filter Group',
+      }),
+    getHelp: () =>
+      i18n.translate('xpack.canvas.uis.arguments.filterGroupLabel', {
+        defaultMessage: 'Create or select a filter group',
+      }),
+    getCreateNewGroup: () =>
+      i18n.translate('xpack.canvas.uis.arguments.filterGroup.createNewGroupLinkText', {
+        defaultMessage: 'Create new group',
+      }),
+  },
+  ImageUpload: {
+    getDisplayName: () =>
+      i18n.translate('xpack.canvas.uis.arguments.imageUploadTitle', {
+        defaultMessage: 'Image upload',
+      }),
+    getHelp: () =>
+      i18n.translate('xpack.canvas.uis.arguments.imageUploadLabel', {
+        defaultMessage: 'Select or upload an image',
+      }),
+    getImageUploading: () =>
+      i18n.translate('xpack.canvas.uis.arguments.imageUpload.imageUploadingLabel', {
+        defaultMessage: 'Image uploading',
+      }),
+    getFileUploadPrompt: () =>
+      i18n.translate('xpack.canvas.uis.arguments.imageUpload.fileUploadPromptLabel', {
+        defaultMessage: 'Select or drag and drop an image',
+      }),
+    getUrlFieldPlaceholder: () =>
+      i18n.translate('xpack.canvas.uis.arguments.imageUpload.urlFieldPlaceholder', {
+        defaultMessage: 'Image {url}',
+        values: {
+          url: URL,
+        },
+      }),
+    urlTypes: {
```

Comment:
nit: I'm not sure I'm a fan of creating an internal, additional namespace... keeping the dictonaries clean makes them easier to type and understand.

Consider `getFileUrlType()`, or perhaps `getUrlTypes()` that returns an object with three keys, rather than three methods within a namespace.